In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
import numpy as np
from scipy.stats import chi2_contingency
from statsmodels.stats import proportion

Question No.1

In [ ]:
cutlet=pd.read_csv("/content/drive/MyDrive/Assignment DS/Assignment 3/Cutlets.csv")
cutlet

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [ ]:
p_value = stats.ttest_ind(cutlet['Unit A'],cutlet['Unit B'])
p_value

TtestResult(statistic=0.7228688704678063, pvalue=0.4722394724599501, df=68.0)

In [ ]:
p_value[1]     # 2-tail probability

0.4722394724599501

Question No.2

In [ ]:
lab=pd.read_csv("/content/drive/MyDrive/Assignment DS/Assignment 3/LabTAT.csv")
lab

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [ ]:
p1=stats.f_oneway(lab['Laboratory 1'],lab['Laboratory 2'],lab['Laboratory 3'],lab['Laboratory 4'])
p1                              #compare the pvalue with α = 0.05

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Inference : Reject Null hypothesis and accept Alternate Hypothesis

Question No.3

In [ ]:
buyer=pd.read_csv('/content/drive/MyDrive/Assignment DS/Assignment 3/BuyerRatio.csv')
buyer

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
ratio=np.array([[50,142,131,70],[435,1523,1356,750]])
ratio

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [ ]:
# Compare p_value with α = 0.05
chi2_contingency(ratio)     ## o/p is (Chi2 stats value, p_value, df, expected obsvations)

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

Inference: As (p-value = 0.6603) > (α = 0.05); Accept the Null Hypothesis.
           Thus, male-female buyer rations are similar across regions and are not related.

In [ ]:
#p1,p2,p3,p4,n1,n2,n3,n4=50,142,131,70,550,351,480,350
#proportion.proportions_ztest()

Question No.4

In [ ]:
data=pd.read_csv('/content/drive/MyDrive/Assignment DS/Assignment 3/Costomer+OrderForm.csv')
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [ ]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [ ]:
order=np.array([[271,267,269,280],[29,33,31,20]])
order               #contingency table

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [ ]:
chi2_contingency(order)     ## o/p is (Chi2 stats value, p_value, df, expected obsvations)

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

Inference: As (p_value = 0.2771) > (α = 0.05); Accept Null Hypothesis.
           Thus, customer order forms defective %  doesnot varies by centre.
           